lib/redlock/client.rb

```diff
@@ -210,10 +211,16 @@ def get_remaining_ttl(resource)
 
       private
 
+      def generate_script_shas
+        @unlock_script_sha = Digest::SHA1.hexdigest(UNLOCK_SCRIPT)
+        @lock_script_sha   = Digest::SHA1.hexdigest(LOCK_SCRIPT)
+        @pttl_script_sha   = Digest::SHA1.hexdigest(PTTL_SCRIPT)
```

Comment:
It's possible to know the exact SHA that Redis will generate for a given script as it also uses `SHA1`.